### PR TITLE
Add OTLP HTTP to gateway

### PR DESCRIPTION
**Pull request by pavolloffay**
Notable changes

- Enables OTLP HTTP by default on gateway 
- microservices and monolithic 
- OTLP HTTP port is always first on the distributor service

### CODE REVIEW

**Comment by pavolloffay** [comment]
My test manifests

```
kubectl apply -f - <<EOF
apiVersion: tempo.grafana.com/v1alpha1
kind: TempoMonolithic
metadata:
  name: sample
spec:
  multitenancy:
    enabled: true
    mode: openshift 
    authentication: 
      - tenantName: dev 
        tenantId: "1610b0c3-c509-4592-a256-a1871353dbfa" 
      - tenantName: prod
        tenantId: "1610b0c3-c509-4592-a256-a1871353dbfb"
  storage:
    traces:
      backend: memory
  resources:
    limits:
      cpu: "500m"
      memory: "2Gi"
  jaegerui:
    enabled: true
    route:
      enabled: true
    resources:
      limits:
        cpu: "500m"
        memory: "1Gi"
EOF

kubectl apply -f - <<EOF
apiVersion: tempo.grafana.com/v1alpha1
kind:  TempoStack
metadata:
  name: simplest
spec:
  managementState: Managed
  storage:
    secret:
      name: minio-test
      type: s3
  storageSize: 1Gi
  resources:
    total:
      limits:
        memory: 2Gi
        cpu: 2000m
  tenants:
    mode: openshift 
    authentication: 
      - tenantName: dev 
        tenantId: "1610b0c3-c509-4592-a256-a1871353dbfa" 
      - tenantName: prod
        tenantId: "1610b0c3-c509-4592-a256-a1871353dbfb"
  template:
    gateway:
      enabled: true 
    queryFrontend:
      jaegerQuery:
        enabled: true
EOF


kubectl apply -f - <<EOF
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: tempostack-traces-reader
rules:
  - apiGroups:
      - 'tempo.grafana.com'
    resources: 
      - dev
      - prod
    resourceNames:
      - traces
    verbs:
      - 'get' 
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: tempostack-traces-reader
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: tempostack-traces-reader
subjects:
  - kind: Group
    apiGroup: rbac.authorization.k8s.io
    name: system:authenticated
EOF


kubectl apply -f - <<EOF
apiVersion: v1
kind: ServiceAccount
metadata:
  name: otel-collector 
  namespace: ploffay
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: tempostack-traces-write
rules:
  - apiGroups:
      - 'tempo.grafana.com'
    resources: 
      - dev
    resourceNames:
      - traces
    verbs:
      - 'create' 
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: tempostack-traces
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: tempostack-traces-write
subjects:
  - kind: ServiceAccount
    name: otel-collector
    namespace: ploffay
EOF


kubectl apply -f - <<EOF
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: cluster-collector
  namespace: ploffay
spec:
  image: ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector-contrib:0.101.0
  mode: deployment
  serviceAccount: otel-collector
  config: |
      extensions:
        bearertokenauth:
          filename: "/var/run/secrets/kubernetes.io/serviceaccount/token"
      receivers:
        otlp:
          protocols:
            http: {}
            grpc: {}
      processors:
        batch: {}
      exporters:
        debug: {}
        otlp/dev:
          endpoint: tempo-simplest-gateway.ploffay.svc.cluster.local:8090
          tls:
            insecure: false
            ca_file: "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
          auth:
            authenticator: bearertokenauth
          headers:
            X-Scope-OrgID: "dev"
        otlphttp/dev:
          endpoint: https://tempo-sample-gateway.ploffay.svc.cluster.local:8080/api/traces/v1/dev
          tls:
            insecure: false
            ca_file: "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
          auth:
            authenticator: bearertokenauth
          headers:
            X-Scope-OrgID: "dev"
      service:
        extensions: [bearertokenauth]
        pipelines:
          traces:
            receivers: [otlp]
            processors: [batch]
            exporters: [debug,otlphttp/dev]
EOF



```

**Comment by IshwarKanse** [comment]
@pavolloffay We need to update the tests/e2e/gateway tests/e2e-openshift/multitenancy/ cases for the change.
```
--- a/tests/e2e-openshift/multitenancy/01-assert.yaml
+++ b/tests/e2e-openshift/multitenancy/01-assert.yaml
@@ -155,6 +155,7 @@ spec:
         - --web.listen=0.0.0.0:8080
         - --web.internal.listen=0.0.0.0:8081
         - --traces.write.otlpgrpc.endpoint=tempo-simplest-distributor.chainsaw-multitenancy.svc.cluster.local:4317
+        - --traces.write.otlphttp.endpoint=https://tempo-simplest-distributor.chainsaw-multitenancy.svc.cluster.local:4318
```
I tested the PR and the changes LGTM, I'm working on adding some additional steps to the tests and will submit the PR after this gets merged.

**Review by IshwarKanse — APPROVED** [review]

File: tests/e2e-openshift/monolithic-multitenancy-openshift/02-assert.yaml [hunk 1]
```diff
@@ -11,10 +11,21 @@ apiVersion: v1
...
   - appProtocol: grpc
-    name: otlp-grpc
+    name: otlp-grpc-grpc
     port: 4317
     protocol: TCP
-    targetPort: 4317
\ No newline at end of file
+    targetPort: 4317
+  - appProtocol: http
+    name: otlp-http-http
```
Comment (andreasgerstmayr) [inline tests/e2e-openshift/monolithic-multitenancy-openshift/02-assert.yaml:23]: Can we use `otlp-http` and `otlp-grpc` as port names?

Comment (pavolloffay) [reply tests/e2e-openshift/monolithic-multitenancy-openshift/02-assert.yaml:23]: These ports names are set by the OTEL operator. They are derived from the receiver name e.g. `otlp/http`

Comment (andreasgerstmayr) [reply tests/e2e-openshift/monolithic-multitenancy-openshift/02-assert.yaml:23]: oh, sorry, I didn't realize this is from the OTEL collector.

**Review by andreasgerstmayr — APPROVED** [review]

**Pull request merged**